### ISSUE TITLE
[bug] conan produces an edge (or maybe outright invalid?) `CMake` grammar that happens to work

### ISSUE THREAD

**Issue by mmomtchev** [conan-io/conan#15867]
### Environment details

* Operating System+version: Ubuntu 22.04
* Compiler+version:
* Conan version: 2.1.0
* Python version: 3.10.12

### Steps to reproduce

```
[requires]
sqlite3/[>=3.45.0]

[generators]
MesonToolchain
PkgConfigDeps
CMakeDeps
```

### Logs

The `CMake` config files contain large amounts of `set_property` containing an `APPEND` as last item:

```
        ########## TARGET PROPERTIES #####################################
        set_property(TARGET SQLite::SQLite3
                     PROPERTY INTERFACE_LINK_LIBRARIES
                     $<$<CONFIG:Release>:${sqlite3_SQLite_SQLite3_OBJECTS_RELEASE}>
                     $<$<CONFIG:Release>:${sqlite3_SQLite_SQLite3_LIBRARIES_TARGETS}>
                     APPEND)

        if("${sqlite3_SQLite_SQLite3_LIBS_RELEASE}" STREQUAL "")
            # If the component is not declaring any "cpp_info.components['foo'].libs" the system, frameworks etc are not
            # linked to the imported targets and we need to do it to the global target
            set_property(TARGET SQLite::SQLite3
                         PROPERTY INTERFACE_LINK_LIBRARIES
                         sqlite3_SQLite_SQLite3_DEPS_TARGET
                         APPEND)
        endif()

        set_property(TARGET SQLite::SQLite3 PROPERTY INTERFACE_LINK_OPTIONS
                     $<$<CONFIG:Release>:${sqlite3_SQLite_SQLite3_LINKER_FLAGS_RELEASE}> APPEND)
        set_property(TARGET SQLite::SQLite3 PROPERTY INTERFACE_INCLUDE_DIRECTORIES
                     $<$<CONFIG:Release>:${sqlite3_SQLite_SQLite3_INCLUDE_DIRS_RELEASE}> APPEND)
        set_property(TARGET SQLite::SQLite3 PROPERTY INTERFACE_LINK_DIRECTORIES
                     $<$<CONFIG:Release>:${sqlite3_SQLite_SQLite3_LIB_DIRS_RELEASE}> APPEND)
        set_property(TARGET SQLite::SQLite3 PROPERTY INTERFACE_COMPILE_DEFINITIONS
                     $<$<CONFIG:Release>:${sqlite3_SQLite_SQLite3_COMPILE_DEFINITIONS_RELEASE}> APPEND)
        set_property(TARGET SQLite::SQLite3 PROPERTY INTERFACE_COMPILE_OPTIONS
                     $<$<CONFIG:Release>:${sqlite3_SQLite_SQLite3_COMPILE_OPTIONS_RELEASE}> APPEND)
```

I don't think this is a valid `CMake` grammar: 
https://cmake.org/cmake/help/latest/command/set_property.html

In `CMake` it happens to work - I don't know if the `APPEND` is not ignored though - however when this target is imported as a `meson` subproject - meson considers - I think rightfully - that `APPEND` is part of the values.

**Comment by memsharded** [comment]
I am doing this test:
```cmake
cmake_minimum_required(VERSION 3.15)

project(test LANGUAGES CXX)

add_library(mytest INTERFACE)

set_property(TARGET mytest PROPERTY INTERFACE_INCLUDE_DIRECTORIES "include1" APPEND)
set_property(TARGET mytest PROPERTY INTERFACE_INCLUDE_DIRECTORIES "include2" APPEND)

get_target_property(mytest_includes mytest INTERFACE_INCLUDE_DIRECTORIES)

message(STATUS "TARGET INCLUDES ${mytest_includes}")
```
It correctly outputs
```
-- TARGET INCLUDES include1;include2
```
If I remove the APPEND:
```cmake
set_property(TARGET mytest PROPERTY INTERFACE_INCLUDE_DIRECTORIES "include1" )
set_property(TARGET mytest PROPERTY INTERFACE_INCLUDE_DIRECTORIES "include2" )
```
Then
```
-- TARGET INCLUDES include2
```

So it seems that CMake is actually accepting the APPEND argument correctly at that position

**Comment by mmomtchev** [comment]
This would require lots of work in the `meson` parser, I would try to get an official statement from `CMake`, because `CMake` dependencies are supposed to work in `meson`.

**Comment by mmomtchev** [comment]
https://gitlab.kitware.com/cmake/cmake/-/issues/25774

**Comment by memsharded** [comment]
Ok, so it seems both, the implementation allows it, so they confirm it works, as proved above, but also it was not intended.

I think we can try to change it, it shouldn't have a high risk.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/15877 merged, this change will be in the next 2.2 release
